static/js/src/services/code.js

```diff
@@ -1,29 +1,52 @@
 'use strict';
 
-function code(commands, blockly) {
-  return { generate, execute };
+function code(commands, blockly, Boards) {
+  return { generate, execute, setWorkshop };
+  var currentWorkshop = null;
+
+  // TODO - this seems pretty wrong. Let's ask Brian or someone who knows Angular
+  // pretty well to take a look!
+  function setWorkshop(workshop) {
+    currentWorkshop = workshop;
+  }
```

Comment:
As far as I can tell, there is nothing about this "service" that needs to be an angular service that needs to be injected.  Instead, it can be constructed more as a stand-alone behavior library (Angular agnostic) in the model section.  When something (like the workshop component) needs a code generator, it can just "new" it up:

```js
this.codeGenerator = new CodeGenerator(workshop);
```

Basically, over time, (since I wrote this code), I've moved to a model where things that have nothing to do with the view (models, data stores, business logic, etc) are not tied to Angular, and are imported using ES2015 import instead.  Injections are for things that need to be stateful, which this "service" does not need to be.

I realize I wrote it this way, but it has been over a year, and my thoughts on this has changed, precisely because of things like this... The Angular 1.x architecture encourages way too much long-term statefulness of things that don't need to be stateful.